kyaml/yaml/rnode.go

```diff
@@ -546,6 +546,29 @@ func (rn *RNode) GetBinaryDataMap() map[string]string {
 	return result
 }
 
+// GetValidatedDataMap retrieves the data map and returns an error if the data
+// map contains entries which are not included in the expected set
+func (rn *RNode) GetValidatedDataMap(expected []string) (map[string]string, error) {
```

Comment:
nits: IMO `expectedKeys` is better and also is consistent with method `validateDataMap`.

Reply:
I've updated the argument name